### PR TITLE
Fix Unicode 15.0 line breaking

### LINKED ISSUE #4146: `icu_segmenter::LineSegmenter` returns breaks other than LB3 and LB7
The following code  using `icu_segmenter = "1.3.2"` returns 3 breaks:
- `LineBreak::Unknown` at 0
- `LineBreak::ZWJ` at 10
- `LineBreak::Ideographic` at 14

```rust
use icu_segmenter::LineSegmenter;

fn main() {
    assert_eq!(
        vec![0, 10, 14],
        LineSegmenter::new_auto()
            .segment_str("🏳️‍🌈")
            .collect::<Vec<_>>()
    );
}
```

According to the [documentation](https://docs.rs/icu_segmenter/latest/icu_segmenter/struct.LineSegmenter.html), the segmenter should only return [LB3] and [LB7], not [LB8a] (`LineBreak::ZWJ`).

[LB3]: https://www.unicode.org/reports/tr14/#LB3
[LB7]: https://www.unicode.org/reports/tr14/#LB7
[LB8a]: https://www.unicode.org/reports/tr14/#LB8a

### PR DESCRIPTION
**Pull request by eggrobin**
The current implementation was attempting the LB25 tailoring recommended in Example 7 of [Section 8.2](https://www.unicode.org/reports/tr14/tr14-49.html#Examples) in UAX14 version 15.0; however, this requires more than one code point of lookahead* because of `(PR | PO) × ( OP | HY )? NU`, which the current implementation of the line segmenter cannot do. Instead this pull request goes back to the untailored LB25 from Unicode 15.0.

The implementation was tested with two million test cases; I last encountered a failure somewhere in the nine thousands. I should probably do an overnight run. Only 200 test cases are included here; as usual, anyone working on the rules should try very long monkey test runs. 

This fixes #4146.

—
\* This will be needed for 15.1 line segmentation too. While we have that capability in the other segmenters, used in the sentence segmenter (the relevant rules are called intermediate match rules or interm(ediate) break states in this implementation), straightforwardly reusing that code would run into into issues as we have so many states in line breaking that we cannot dedicate a whole bit to that property of the state. This can probably be worked around (as far as I can tell we use the sign bit for a property of two special states, so we could probably be a bit more sparing), but will come later.

### CODE REVIEW

**Comment by eggrobin** [comment]
The tests are failing because LineBreakTest.txt tests for LB25 as tailored by Example 7, not for the untailored version.
I know how that particular sausage is made, so I’ll head over to unicode-org/unicodetools to generate a LineBreakTest.txt for the untailored line breaking algorithm…

File: components/segmenter/tests/testdata/LineBreakTest.txt [hunk 1]
```diff
@@ -1,12 +1,9 @@
-# LineBreakTest-15.0.0.txt
-# Date: 2022-02-26, 00:38:39 GMT
-# © 2022 Unicode®, Inc.
+# THIS IS NOT LineBreakTest-15.0.0.txt
```
Comment (robertbastian) [inline components/segmenter/tests/testdata/LineBreakTest.txt:1]: Are these manual changes to the test file? Please document its provenance.

Comment (eggrobin) [reply components/segmenter/tests/testdata/LineBreakTest.txt:1]: Documented (though you won’t like it :-p).

File: components/segmenter/tests/testdata/LineBreakExtraTest.txt [hunk 2]
```diff
@@ -12,3 +12,211 @@
 × 0031 × 002F × 0020 ÷ # × [NU] × [SY] × [SP] ÷
 × 1F1E6 × 1F1E6 × 0020 ÷ # × [RI] × [RI] × [SP] ÷
 × 05D0 × 002D × 0020 ÷ # × [HL] × [HY] × [SP] ÷
+# The following 200 monkey tests were generated from
+# https://github.com/eggrobin/icu/tree/export-monkeys-15.0-untailor-lb
+# (specifically, at ea318304775e0a5194785f51120672aafac7b2bd)
```
[the remark this answers is not in the document]
Comment (eggrobin) [reply components/segmenter/tests/testdata/LineBreakExtraTest.txt:17]: Of course the change that makes it possible to export the monkey tests will be checked in, as part of https://github.com/unicode-org/icu/pull/2637 (whose reviewers I really need to poke); but that will be on top of 15.1.

[the remark this answers is not in the document]
Comment (eggrobin) [reply components/segmenter/tests/testdata/LineBreakExtraTest.txt:17]: Sadly that is not really feasible, as discussed.

**Review by robertbastian — APPROVED** [review]
🏳️‍🌈🏳️‍🌈🏳️‍🌈

**Comment by robertbastian** [comment]
this is changelog worthy

**Pull request merged**